src/js/requestHandlers.js

```diff
@@ -230,6 +235,85 @@ sjrk.storyTelling.server.deleteStoryFromCouch = function (id, deleteStoryDataSou
     return promise;
 };
 
+sjrk.storyTelling.server.deleteStoryFiles = function (storyContent, serverConfig) {
+    var filesToDelete = [];
+
+    fluid.each(storyContent, function (block) {
+        var blockFileName = "";
+
+        if (block.blockType === "image") {
+            blockFileName = block.imageUrl;
+        } else if (block.blockType === "audio" || block.blockType === "video") {
+            blockFileName = block.mediaUrl;
+        }
+
+        if (sjrk.storyTelling.server.isValidMediaFilename(blockFileName)) {
+            filesToDelete.push(blockFileName);
+        }
+    });
+
+    // remove duplicate entries so we don't try to delete already-deleted files
+    filesToDelete = filesToDelete.filter(function (fileName, index, self) {
+        return self.indexOf(fileName) === index;
+    });
+
+    fluid.each(filesToDelete, function (fileToDelete) {
+        sjrk.storyTelling.server.deleteSingleFileRecoverable(fileToDelete, serverConfig);
+    });
+};
+
+sjrk.storyTelling.server.isValidMediaFilename = function (fileName, serverConfig) {
+    try {
+        if (!fileName || fileName === "null" || fileName === "undefined")
+        {
+            throw "The filename is not valid: not truthy.";
+        }
+
+        if (sjrk.storyTelling.server.isValidUuidFileName(fileName)) {
+            throw "The filename is not valid: does not match UUID file name pattern.";
+        }
+
+        var fileWithFullPath = sjrk.storyTelling.server.getServerPathForFile(fileName, serverConfig.uploadedFilesHandlerPath);
+        fs.accessSync(fileWithFullPath, fs.constants.W_OK | fs.constants.R_OK);
+
+        return true; // it is a valid file name
+    } catch (err) {
+        return false; // it is not a valid file name
+    }
+};
+
+sjrk.storyTelling.server.isValidUuidFileName = function (fileName) {
```

Comment:
The name does not match the function -- it's the inverse.

Reply:
Though, I think once `isValidMediaFilename` is simplified, we can fold this function's behaviour directly into that function. And remove this separate function.